code/obj/submachine/seed.dm

```diff
@@ -1084,26 +1084,110 @@
 	attack_ai(mob/user as mob)
 		return src.Attackhand(user)
 
-	attack_hand(var/mob/user as mob)
-		src.add_dialog(user)
-		var/dat = "<B>[src.name]</B><BR><HR>"
-		dat += "<b>Amount to Vend</b>: <A href='?src=\ref[src];amount=1'>[src.vendamt]</A><br>"
-		if (src.category)
-			dat += "<b>Filter</b>: [src.category] <A href='?src=\ref[src];category=1'>(Clear)</A><br>"
-		else
-			dat += "<b>Filter</b>: <A href='?src=\ref[src];category=1'>(Set)</A><br>"
-		if (!src.can_vend)
-			dat+= "<u>Unit currently out of charge. Please wait.</u><br>"
-		dat += "<br>"
+	proc/getPlantImg(var/datum/plant/plant)
+		if(!istype(plant))
+			return
+		var/path = plant.type
+		var/img = plant_base64_cache[path]
+		if(!img)
+			var/icon/result_icon
+			if(plant.crop)
+				var/atom/crop = plant.crop
+				result_icon = icon(initial(crop.icon), initial(crop.icon_state), frame=1)
+			else if(plant.plant_icon)
+				if(plant.sprite)
+					result_icon = icon(plant.plant_icon, "[plant.sprite]-G4", frame=1)
+				else if(plant.override_icon_state)
+					result_icon = icon(plant.plant_icon, "[plant.override_icon_state]-G4", frame=1)
+				else
+					result_icon = icon(plant.plant_icon, "[plant.name]-G4", frame=1)
+
+			if(result_icon)
+				img = icon2base64(result_icon)
+				plant_base64_cache[path] = img
```

Comment:
Should probably have a default case, i.e.
```suggestion
			if (!result_icon)
				result_icon = // some default
			img = icon2base64(result_icon)
			plant_base64_cache[path] = img
```

Reply:
Returns an empty string instead.
Since it's not null, it prevents the proc from attempting to cache the image again, since I just added an `isnull()` check.
Since it's falsy, this lets tgui handle the missing image by replacing it with an icon.
![image](https://user-images.githubusercontent.com/45642475/155050205-8fb20570-9aa6-46cd-be81-c7cca847297d.png)